### PR TITLE
Add empty_parentheses_with_trailing_closure rule

### LINKED ISSUE #885: Parenthesis in trailing closure rule
When using trailing closures, one can add `()` after the method call, but that should be discouraged.

```swift
let x = [1, 2]

// shouldn't trigger
_ = x.map { $0 + 1 }

// should trigger
_ = x.map() { $0 + 1 }
```

Note that there are cases when parenthesis are allowed (when there're more than 1 parameter):

```swift
// shouldn't trigger
_ = x.reduce(0) { $0 + $1 }
```

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes #885

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/EmptyParenthesesWithTrailingClosureRule.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        ],
+        triggeringExamples: [
+            "[1, 2].map↓() { $0 + 1 }",
+            "[1, 2].map↓( ) { $0 + 1 }\n",
+            "[1, 2].map↓() { number in\n number + 1 \n}\n",
+            "[1, 2].map↓(  ) { number in\n number + 1 \n}\n"
+        ]
+    )
+
+    public enum Kind: String {
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/EmptyParenthesesWithTrailingClosureRule.swift]: this is the second place where this enum is needed and the 3rd where `"source.lang.swift.expr.call"` is used. Should we create a shared enum? If so, where?

File: Source/SwiftLintFramework/Rules/EmptyParenthesesWithTrailingClosureRule.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        guard let offset = (dictionary["key.offset"] as? Int64).flatMap({ Int($0) }),
+            let length = (dictionary["key.length"] as? Int64).flatMap({ Int($0) }),
+            let nameOffset = (dictionary["key.nameoffset"] as? Int64).flatMap({ Int($0) }),
+            let nameLength = (dictionary["key.namelength"] as? Int64).flatMap({ Int($0) }),
+            let bodyLength = (dictionary["key.bodylength"] as? Int64).flatMap({ Int($0) }),
+            bodyLength > 0 else {
+                return []
+        }
+
+        let rangeStart = nameOffset + nameLength
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/EmptyParenthesesWithTrailingClosureRule.swift]: using bodyOffset and bodyLength were giving incorrect ranges ¯\_(ツ)_/¯

File: Source/SwiftLintFramework/Rules/EmptyParenthesesWithTrailingClosureRule.swift (hunk 1 above)
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/EmptyParenthesesWithTrailingClosureRule.swift]: Yes! This should live in SourceKittenFramework though.

**Comment by marcelofabri** [comment]
🎉 🚢

**Pull request merged**